src/blockdev.rs

```diff
@@ -225,6 +225,22 @@ pub fn try_discard_all(file: &mut File) -> Result<bool> {
 ioctl_none!(blkrrpart, 0x12, 95);
 ioctl_write_ptr_bad!(blkdiscard, request_code_none!(0x12, 119), [u64; 2]);
 
+pub fn udev_settle_until_partition_appears(device: &str, label: &str) -> Result<()> {
+    eprint!("Waiting for partition label {} to come online", label);
+    for _ in (0..10).rev() {
+        udev_settle()?;
+
+        let dev = get_partition_with_label(device, label)?;
+        if dev.is_some() {
+            return Ok(());
+        }
+
+        sleep(Duration::from_millis(100));
+    }
+
+    return Err("timed out waiting for boot partition to show up".into());
```

Comment:
We're waiting a total of 1s?  Maybe add that to the error; something like
`
let maxwait = Duration::from_secs(1);
for dur in (0..maxwait/10) {
   ...
   sleep(dur)
}
return Err("timed out waiting {} for partition label '{}' of device {}", maxwait, label, device);
```